### PR TITLE
Update mobile layout for governance

### LINKED ISSUE #160: [Mobile]: layout governance page
use figma!

this includes, proposals, consensus and treasury transaction

Governance - proposal (2)

### PR DESCRIPTION
**Pull request by JustinBeBoy**
Resolve #160 

**Screenshot**

image

image

image

### CODE REVIEW

File: ui/cryptomaterial/dropdown.go [hunk 1]
```diff
@@ -149,6 +151,17 @@ func (d *DropDown) SelectedIndex() int {
...
+	d.convertTextSize = fun
+}
+
+func (d *DropDown) getTextSize(textSize unit.Sp) unit.Sp {
+	if d.convertTextSize == nil {
+		return textSize
+	}
+	return d.convertTextSize(textSize)
+}
+
```
Comment (dreacot) [inline ui/cryptomaterial/dropdown.go:164]: i don't see any conversion actually happening, am i missing something?

File: ui/cryptomaterial/util.go [hunk 2]
```diff
@@ -139,20 +139,26 @@ func AnyKeyWithOptionalModifier(modifier key.Modifiers, keys ...string) key.Set
 	return key.Set(keysWithModifier)
 }
 
-func UniformPadding(gtx layout.Context, body layout.Widget) layout.Dimensions {
+func UniformPadding(gtx layout.Context, body layout.Widget, isMobileView ...bool) layout.Dimensions {
+	_isMobileView := len(isMobileView) > 0 && isMobileView[0]
```
Comment (dreacot) [inline ui/cryptomaterial/util.go:143]:
```suggestion
	isMobileView := len(isMobileView) > 0 && isMobileView[0]
```

File: ui/page/governance/proposal_details_page.go [hunk 3]
```diff
@@ -474,40 +553,62 @@ func (pg *ProposalDetails) layoutDescription(gtx C) D {
...
 			)
 		},
 		pg.layoutRedirect(values.String(values.StrViewOnPoliteia), pg.redirectIcon, pg.viewInPoliteiaBtn),
 		pg.lineSeparator(layout.Inset{Top: values.MarginPadding16, Bottom: values.MarginPadding16}),
 	}
 
-	_, ok := pg.proposalItems[proposal.Token]
-	if ok {
-		w = append(w, pg.proposalItems[proposal.Token].widgets...)
+	// _, ok := pg.proposalItems[proposal.Token]
```
Comment (dreacot) [inline ui/page/governance/proposal_details_page.go:604]: remove comment?

File: ui/page/governance/treasury_page.go [hunk 4]
```diff
@@ -213,19 +214,24 @@ func (pg *TreasuryPage) layout(gtx C) D {
...
 					return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
 						layout.Rigid(func(gtx C) D {
-							return layout.Flex{Axis: layout.Horizontal}.Layout(gtx,
-								layout.Rigid(pg.Theme.Label(values.TextSize20, values.String(values.StrTreasurySpending)).Layout),
+							return layout.Flex{Alignment: layout.Middle}.Layout(gtx,
+								layout.Rigid(pg.Theme.Label(pg.ConvertTextSize(values.TextSize20), values.String(values.StrTreasurySpending)).Layout),
 								layout.Rigid(func(gtx C) D {
-									return layout.Inset{Top: values.MarginPadding3}.Layout(gtx, pg.infoButton.Layout)
+									return pg.infoButton.Layout(gtx)
+									// return layout.Inset{Top: values.MarginPadding2}.Layout(gtx, pg.infoButton.Layout)
```
Comment (dreacot) [inline ui/page/governance/treasury_page.go:234]: remove comment

File: ui/renderers/markdown.go [hunk 5]
```diff
@@ -163,6 +164,7 @@ func (p *MarkdownProvider) renderListItem(content string) {
 		lbl := p.getLabel()
 		strongLabel := p.getLabel()
 		strongLabel.Font.Weight = font.Bold
+		fmt.Println("-------getLabel----222222-->", lbl.TextSize)
```
Comment (dreacot) [inline ui/renderers/markdown.go:167]: remove debug comment

File: ui/renderers/markdown.go [hunk 6]
```diff
@@ -271,6 +278,9 @@ func (p *MarkdownProvider) renderBlock() {
 
 func (p *MarkdownProvider) getLabel() cryptomaterial.Label {
 	lbl := p.theme.Body1("")
+	size := p.ConvertTextSize(values.TextSize16)
+	// fmt.Println("-------getLabel------>", size)
```
Comment (dreacot) [inline ui/renderers/markdown.go:282]: remove comment

File: ui/renderers/markdown.go [hunk 7]
```diff
@@ -214,6 +220,7 @@ func (p *MarkdownProvider) prepareLink(_ /*node*/ *ast.Link, _ /*entering*/ bool
 }
 
 func (p *MarkdownProvider) renderBlock() {
+	fmt.Println("------renderBlock---->")
```
Comment (dreacot) [inline ui/renderers/markdown.go:223]: remove comment

File: ui/renderers/markdown.go [hunk 8]
```diff
@@ -289,6 +299,7 @@ func (p *MarkdownProvider) getLabel() cryptomaterial.Label {
 
 func (p *MarkdownProvider) render(content *strings.Builder) {
 	lbl := p.getLabel()
+	fmt.Println("-------getLabel----111111-->", lbl.TextSize)
```
Comment (dreacot) [inline ui/renderers/markdown.go:302]: remove comment

File: ui/window.go [hunk 9]
```diff
@@ -64,9 +64,9 @@ type WriteClipboard struct {
 func CreateWindow(mw *libwallet.AssetsManager, version string, buildDate time.Time) (*Window, error) {
 	appTitle := giouiApp.Title(values.String(values.StrAppName))
 	// appSize overwrites gioui's default app size of 'Size(800, 600)'
-	appSize := giouiApp.Size(values.AppWidth, values.AppHeight)
+	appSize := giouiApp.Size(values.MobileAppWidth, values.MobileAppHeight)
 	// appMinSize is the minimum size the app.
-	appMinSize := giouiApp.MinSize(values.AppWidth, values.AppHeight)
+	appMinSize := giouiApp.MinSize(values.MobileAppWidth, values.MobileAppHeight)
```
Comment (dreacot) [inline ui/window.go:69]: revert

**Review by dreacot — CHANGES_REQUESTED** [review]
1. the filter icon doesn't look center aligned with the drodown

Screenshot from 2023-12-12 10-29-02

2. when filtering is not selected, both the filter dropdowns and search bar should be hidden.

3. the filter icon colour should be our primary blue icon when selected, and should be grey when not selected.
see similar colour in the screenshot
Screenshot from 2023-12-12 10-34-00
Screenshot from 2023-12-12 10-33-49.

for the colour see figma https://www.figma.com/file/MF6UCOtuX22Mk0ji1DnO6J/Version-3?type=design&node-id=3437-94761&mode=design&t=tmunnHys1sALKMdX-0

4. the proposal details page is distorted
Screenshot from 2023-12-12 10-31-39

File: ui/values/dimensions.go [hunk 10]
```diff
@@ -65,10 +65,12 @@ var (
 	MarginPadding75       = unit.Dp(75)
 	MarginPadding80       = unit.Dp(80)
 	MarginPadding96       = unit.Dp(96)
+	DP85                  = unit.Dp(85)
```
Comment (dreacot) [inline ui/values/dimensions.go:68]:
```suggestion
	MarginPadding85                  = unit.Dp(85)
```

right now it was confusing for me when i was trying to make some changes in my PR, a single refactor should convert everything at a later time, for now let's keep using margin, padding

File: ui/cryptomaterial/dropdown.go (hunk 1 above)
Comment (JustinBeBoy) [reply ui/cryptomaterial/dropdown.go:164]: It actually happens, but first you need set convert function for it

File: ui/cryptomaterial/util.go (hunk 2 above)
Comment (JustinBeBoy) [reply ui/cryptomaterial/util.go:143]: isMobileView is user on param of function

File: ui/values/dimensions.go (hunk 10 above)
Comment (JustinBeBoy) [reply ui/values/dimensions.go:68]: okay

**Comment by JustinBeBoy** [comment]
> 1. the filter icon doesn't look center aligned with the drodown
> 
> Screenshot from 2023-12-12 10-29-02
> 
> 2. when filtering is not selected, both the filter dropdowns and search bar should be hidden.
> 3. the filter icon colour should be our primary blue icon when selected, and should be grey when not selected.
>    see similar colour in the screenshot
>    Screenshot from 2023-12-12 10-34-00
>    Screenshot from 2023-12-12 10-33-49.
> 
> for the colour see figma https://www.figma.com/file/MF6UCOtuX22Mk0ji1DnO6J/Version-3?type=design&node-id=3437-94761&mode=design&t=tmunnHys1sALKMdX-0
> 
> 4. the proposal details page is distorted
>    Screenshot from 2023-12-12 10-31-39

done

**Review by dreacot — APPROVED** [review]

**Pull request merged**